src/layers/basic.jl

```diff
@@ -172,31 +172,45 @@ function Base.show(io::IO, l::Dense)
 end
 
 """
-    Diagonal(α, β)
-    Diagonal(size::Integer...)
+    Diagonal(scale, bias)
+    Diagonal(size::Integer...; bias = true, init = ones32)
```

Comment:
To match Dense etc. I'd put the method which takes integers first:
```suggestion
    Diagonal(size::Integer...; bias = true, init = ones32)
    Diagonal(scale::AbstractArray, [bias])
```